model.py

```diff
@@ -49,11 +48,15 @@ def forward(self, x):
         q = q.view(B, T, self.n_head, C // self.n_head).transpose(1, 2) # (B, nh, T, hs)
         v = v.view(B, T, self.n_head, C // self.n_head).transpose(1, 2) # (B, nh, T, hs)
 
-        # causal self-attention; Self-attend: (B, nh, T, hs) x (B, nh, hs, T) -> (B, nh, T, T)
-        att = (q @ k.transpose(-2, -1)) * (1.0 / math.sqrt(k.size(-1)))
-        att = att.masked_fill(self.bias[:,:,:T,:T] == 0, float('-inf'))
-        att = F.softmax(att, dim=-1)
-        att = self.attn_dropout(att)
+        # the output of sdp = (batch, num_heads, seq_len, head_dim)
+        y = torch.nn.functional.scaled_dot_product_attention(
```

Comment:
super nit: no need to change but
`y = F.scaled_dot_product_attention(q, k, v, None, self.dropout_p, is_causal=True)` looks alot for economically and could be a little more enticing?

Reply:
agreed!